sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_utils.py

```diff
@@ -15,12 +15,59 @@
 from ._generated.models import (
     CommunicationIdentifierModel,
     CommunicationUserIdentifierModel,
-    PhoneNumberIdentifierModel
+    PhoneNumberIdentifierModel,
+    CallLocator
 )
+if TYPE_CHECKING:
+    from ._models import ServerCallLocator, GroupCallLocator
 
-def serialize_identifier(
-        identifier:CommunicationIdentifier
-        ) -> Dict[str, Any]:
+
+def build_call_locator(
```

Comment:
Can we use overload for this method to only accept

call_locator or *, group_call_id or *, server_call_id?

Reply:
I want to avoid duplicating the args/kwargs check in both the sync and async clients, which is why I refactored into a single method here because the code is kinda fussy.
I'll play with it more to see if I can simplify this code by doing the kwarg-popping at the method call instead.